configure.ac

```diff
@@ -235,14 +235,14 @@ if test "x${debug}" = xyes; then
       CXXFLAGS="$CXXFLAGS -g"
    fi
    CXXFLAGS=`echo $CXXFLAGS | ${SED} 's/-O. *//g' | ${SED} 's/-fast *//g'`
-   CXXFLAGS="$CXXFLAGS -O0"
+   CXXFLAGS="$CXXFLAGS -O0 -Wall -Wextra"
```

Comment:
Use of "-Wall -Wextra" should be an option added by users.
Please remove.

Reply:
> Use of "-Wall -Wextra" should be an option added by users.
> Please remove.

fixed

Reply:
Adding E3SM_IO_UNUSED to avoid warnings from -Wextra
is overdoing it. No software would go that extra mile.
It also makes the codes messy and reader unfriendly.

Reply:
> Adding E3SM_IO_UNUSED to avoid warnings from -Wextra
> is overdoing it. No software would go that extra mile.
> It also makes the codes messy and reader unfriendly.

That's how HDF did.
Should we leave the warning open? There is no other way to suppress the warning.

Reply:
Just fix the warnings from -Wall.
This should not need to add E3SM_IO_UNUSED.
Try and see.

Reply:
fixed

Reply:
resubmit a clean PR and close this one